src/evict/evict_page.c

```diff
@@ -420,7 +420,7 @@ __evict_review(
 		 * fallen behind current.
 		 */
 		if (modified)
-			WT_RET(__wt_txn_update_oldest(session, false));
+			WT_RET(__wt_txn_update_oldest(session, true, false));
```

Comment:
This change is arguable -- before evicting a page, be strict about updating the oldest ID but don't wait for a lock on the snapshot table.  I could go either way on this one based on further perf testing, but I think this change is safe and better than what we were doing before.